apps/arena/lib/arena/game_tracker.ex

```diff
@@ -144,6 +155,37 @@ defmodule Arena.GameTracker do
     put_in(data, [:players, player_id, :bounty_quest_id], bounty_quest_id)
   end
 
+  defp generate_player_result(nil, _player_id) do
+    %{}
+  end
+
+  defp generate_player_result(match_data, player_id) do
+    duration = System.monotonic_time(:millisecond) - match_data.start_at
+
+    player_data =
+      Map.get(match_data.players, player_id)
+
+    %{
```

Comment:
Can we extract this and the one in `generate_results/2` so we don't need to modify the same map in 2 places?

Reply:
sure thing!